src/Query/BookingQueryInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace RebelCode\Diary\Query;
+
+/**
+ * Any object that can describe a bookings query.
+ *
+ * @since [*next-version*]
+ */
+interface BookingQueryInterface
+{
+    /**
+     * Gets the query condition.
+     *
+     * @return \Dhii\Espresso\BooleanExpressionInterface
+     */
+    public function getCondition();
```

Comment:
It seems to me that this will need to have specialized handling in the implementation nonetheless. For example, I can have an expression like e.g. `BooleanExpression implements BooleanExpressionInterface`, which simply represents a `true` or `false` value. This probably needs to be interpreted as e.g. `WHERE 1`. Or maybe, I have a expression like `$myFancyExpr = new ServerStatusCondition('up');`, where `ServerStatusCondition implements BooleanExpressionInterface` retrieves a status code from a remote API and compares it to a given status code. How would this then be translated into, say, a query string?
In the light of this problem, I suggest that another, diary-specific interface is put in that would represent a specific subset of conditions. The name could be something like `ConditionInterface`.